### DISCUSSION TITLE
Tests Directory with Google Colab

### DISCUSSION THREAD

**Discussion by vladtkachuk4** [ucbds-infra/otter-grader#860] [Q&A] [answered]
I am planning to use otter-grader with Google Colab for an assignment. 
I have worked through the Tutorial in the Docs. Using the files provided in the tutorial (and removing the non-autograded questions) I've been able to generate a Colab notebook by adding `runs_on: colab` to the Assignment Config and running `otter assign`. When I upload the student notebook to Colab and run it, the following line throws an error since it is expecting a `tests` directory:
`grader = otter.Notebook(colab=True)`
This is of course expected. I've been able to get things to work by adding the following code cell:
```
from google.colab import drive
drive.mount('/content/drive')
!cp -r /content/drive/MyDrive/otter-grader/tests ./test
```

What I am wondering is what the best way to distribute the student notebooks would be with Google Colab such that students also have access to the public tests. I don't want them to need to add the tests manually to their drive and then mount and copy it like I have.

Another minor note: I've had to manually add `%pip install -q otter-grader` as the first code cell to the student generated notebook. I'm assuming there is some Assignment Config setting that should cause for this to automatically be done? I suspected that `runs_on: colab` would automatically do this, but it doesn't.

**Comment by chrispyles** [comment, marked as answer]
> What I am wondering is what the best way to distribute the student notebooks would be with Google Colab such that students also have access to the public tests. I don't want them to need to add the tests manually to their drive and then mount and copy it like I have.

I think the easiest thing to do would be to put the in a github repo, clone it at the top of the notebook, and then `os.chdir` into the repo directory.

> Another minor note: I've had to manually add %pip install -q otter-grader as the first code cell to the student generated notebook. I'm assuming there is some Assignment Config setting that should cause for this to automatically be done? I suspected that runs_on: colab would automatically do this, but it doesn't.

This is a good FR, can you open an issue for it?

**Reply by vladtkachuk4** [reply]
Thanks and done.
